### PR TITLE
Use parallel instead of xargs (#391)

### PR DESCRIPTION
**Pull request by AladW**
Solves #391

### CODE REVIEW

File: lib/aur-sync [hunk 1]
```diff
@@ -333,11 +333,11 @@ fi
 cd_safe "$aur_workdir"
 
 if ((download)); then
-    xargs -a "$tmp"/queue aur fetch "${fetch_args[@]}"
+    parallel -X aur fetch "${fetch_args[@]}" :::: "$tmp"/queue
```
Comment (AladW) [inline lib/aur-sync]: I'm not sure why, but in the past I've seen that multiple jobs when cloning over SSH was problematic. Needs testing

File: lib/aur-sync [hunk 2]
```diff
@@ -357,7 +357,7 @@ if ((build)); then
     build_args+=(-d "$repo" -r "$root" -a "$tmp"/queue)
 
     # check if dependency graph is valid
-    printf '%s\0' "$tmp_view"/*/.SRCINFO | xargs -0 aur graph >/dev/null
+    aur graph "$tmp_view"/*/.SRCINFO >/dev/null
```
Comment (AladW) [inline lib/aur-sync]: I went with a simple glob here to avoid an additional parallel invocation... I guess we're pretty unlikely to reach ARG_MAX here anyway.

File: lib/aur-vercmp [hunk 3]
```diff
@@ -58,18 +58,12 @@ my_vercmp() {
...
-    }'
-}
-
 parse_aur() {
     aur rpc -t info | jq -r '.results[] | [.Name, .Version] | @tsv'
 }
 
 parse_sync() {
-    pacman -Slq "$1" | prefix "$1" | xargs -r pacman -Sddp --print-format '%n %v'
+    pacman -Slq "$1" | parallel -rXj1 pacman -Sddp --print-format '%n %v' "$1"/{}
```
Comment (AladW) [inline lib/aur-vercmp]: Note that `parallel` conveniently copies the "$1"/ prefix with `-X`, removing the need for a separate function.

File: makepkg/PKGBUILD [hunk 4]
```diff
@@ -10,13 +10,11 @@ source=('git+https://github.com/AladW/aurutils')
 sha256sums=('SKIP')
 conflicts=('aurutils')
 provides=('aurutils')
-depends=('pacman>=5.0' 'git' 'jq' 'pacutils' 'wget')
+depends=('pacman>=5.0' 'git' 'jq' 'pacutils' 'wget' 'parallel')
```
Comment (AladW) [inline makepkg/PKGBUILD]: A possible place to add the `will cite` notice could be an `.install` file. That or a forum post when 2.0.0 is announced. It seems out of scope to be include this in `aur(1)` or a different manual.

Comment (rafasc) [reply makepkg/PKGBUILD]: Are you referring something like just a warning or actually doing 
`printf 'will cite\n' | parallel --citation` as part of the `.install`?

I am not a lawyer, but:
Based on how the aur works, the commands in `.install` are technically run by the user, so I guess it could be an option. Especially if we use a custom `$PARALLEL_HOME` for aurutils, in order to isolate aurutils use of the tool.

Comment (AladW) [reply makepkg/PKGBUILD]: In this it would be a warning. If we wanted to run this automatically, it should be part of the prepare() function or such (since it needs to be run as the user running the PKGBUILD), though I guess this entails an unattended modification of the home directory and would thus be frowned upon.

I like the idea of using a custom `$PARALLEL_HOME`. Perhaps we could add a small wrapper to `lib` that sets it both to `~/.config/aurutils/parallel` and wraps any `parallel` commands with `--will-cite` (leaving a comment as to the licensing problematic)

**Comment by AladW** [comment]
I added `aur-jobs` which does what https://github.com/AladW/aurutils/pull/394#discussion_r189447439 describes. Unless there's any weird bugs I don't know I think this can be closed.
